include/seqan3/io/alignment_file/input.hpp

```diff
@@ -846,7 +846,7 @@ class alignment_file_input
     //!\privatesection
 
     //!/brief Initialisation based on a filename.
-    void init(std::filesystem::path & filename)
+    void init_by_filename(std::filesystem::path & filename)
```

Comment:
why is the path captured by reference? 
The constructor doesn't do anything with this right?
So should it not be `const &`?

Reply:
oh I actually do something. within `make_secondary_stream` the filename might be stripped from its compression extension!

Reply:
but then it should be a copy. You only want to strip it internally I guess?

Reply:
It is already copied by the constructor so it's safe to capture it by reference in the `init` function?

Reply:
But then someone modifies the constructor and does not know that the variable was changed. So this function should make clear, that it does not change the path or it documents that it might.

Reply:
Alright. Any preference?

Reply:
By copy and then move it in from the calling function would be my preference.

Reply:
ok

Reply:
done